### DISCUSSION TITLE
Align `capitalize` with the built-in `Capitalize<T>` TypeScript type

### DISCUSSION THREAD

**Discussion by aleclarson** [radashi-org/radashi#47] [Ideas]
[As I mentioned](https://github.com/sodiray/radash/pull/414#issuecomment-2192717455) on @rafawendel's [Radash PR](https://github.com/sodiray/radash/pull/414), the current `capitalize` differs in behavior from TypeScript's `Capitalize<T>` built-in type.

> The built-in `Capitalize` type does not match the behavior of Radash `capitalize`.
> 
> ```ts
> type Foo = Capitalize<'foo Bar'>
> //   ^? 'Foo Bar'
> 
> let foo = _.capitalize('foo Bar')
> // => 'Foo bar'
> ```

I suggest we change Radashi's `capitalize` to align with it.

**Comment by localusercamp** [comment]
Agree. 👍 
`_.capitalize` do not "capitalize" it do "lowercase all and capitalize" which is wierd.

**Comment by stefaanMLB** [comment]
Agree

**Comment by aleclarson** [comment]
One downside is we won't be matching Lodash's `capitalize`. 😕 

https://docs-lodash.com/v4/capitalize/

**Reply by aleclarson** [reply]
We might want to just implement `upperFirst` instead 🤔 

https://docs-lodash.com/v4/upper-first/
